dev/reference/error-codes.md

```diff
@@ -18,13 +18,103 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
 | 8003 | `ADMIN CHECK TABLE` 命令在遇到行数据跟索引不一致的时候返回该错误 |
 | 8004 | 单个事务过大，原因及解决方法请参考[这里](/dev/faq/tidb.md#433-transaction-too-large-是什么原因怎么解决) |
 | 8005 | 事务在 TiDB 中遇到了写入冲突，原因及解决方法请参考[这里](/dev/faq/tidb.md#九故障排除) |
+| 8018 | 插件无法重新载入，原因是没有载入过 |
+| 8019 | 重新载入的插件版本与之前不同，无法重新载入 |
+| 8020 | 表被锁 |
+| 8021 | key 不存在 |
+| 8022 | 事务提交失败，但可以进行重试 |
+| 8023 | 不能设置空值 |
+| 8024 | 非法的事务 |
+| 8025 | 写入的单条键值对过大 |
+| 8026 | 没有实现的接口 |
+| 8027 | 表结构版本过期 |
+| 8028 | 表结构发生了变化 |
+| 8029 | 错误值 |
+| 8030 | 转变为带符号正整数发生了越界，显示为负数 |
+| 8031 | 负数转变为无符号数转变为正数 |
+| 8032 | 非法的 year 格式 |
+| 8033 | 非法的 year 值 |
+| 8034 | 不正确的 datetime 值 |
+| 8036 | 非法的 time 格式 |
+| 8037 | 非法的 week 格式 |
+| 8038 | 字段无法获取到默认值 |
+| 8039 | 索引的偏移量超出范围 |
+| 8042 | 表结构的状态为不存在 |
+| 8043 | 列信息的状态为不存在 |
+| 8044 | 索引的状态为不存在 |
+| 8045 | 非法的表数据 |
+| 8046 | 列信息的状态为不可见 |
+| 8047 | 设置的系统变量值并没有支持，通常在用户设置了数据库不支持的变量值的告警信息里出现 |
+| 8048 | 设置了不支持的数据库隔离级别 |
+| 8049 | 载入权限相关表失败 |
+| 8050 | 设置了不支持的权限类型 |
+| 8051 | 未知的字段类型 |
+| 8052 | 来自客户端的数据包的序列号错误 |
+| 8053 | 获取到了非法的自增列值 |
+| 8055 | 当前快照过旧，数据可能已经被 GC |
+| 8056 | 非法的表 ID |
+| 8057 | 非法的字段类型 |
+| 8058 | 申请了不存在的自动变量类型 |
+| 8059 | 获取自动随机量失败 |
+| 8060 | 非法的自增列偏移量 |
+| 8061 | 不支持的 SQL Hint |
+| 8062 | SQL Hint 中使用了非法的 token，与 Hint 的保留字冲突 |
+| 8063 | SQL Hint 中限制内存使用量超过系统设置的上限，忽略该设置 |
+| 8064 | 解析 SQL Hint 失败 |
+| 8065 | SQL Hint 中使用了非法的整数 |
+| 8066 | JSON_OBJECTAGG 函数的第二个参数是非法参数 |
+| 8101 | 插件 ID 格式错误，正确的格式是 [name]-[version] 并且 name 和 version 中不能带有 '-' |
+| 8102 | 无法读取插件定义信息 |
+| 8103 | 插件名称错误 |
+| 8104 | 插件版本错误 |
+| 8105 | 插件被重复载入错误 |
+| 8106 | 插件定义的系统变量名称没有以插件名作为开头 |
+| 8107 | 载入的插件版本太低 |
+| 8108 | 不支持的执行计划类型 |
+| 8109 | analyze 索引时找不到指定的索引 |
+| 8110 | 不能进行笛卡尔积，需要将配置文件里 cross-join 设置为 true |
+| 8111 | execute 语句执行时找不到对应的 prepare 语句 |
+| 8112 | execute 语句的参数个数与 prepare 语句不符合 |
+| 8113 | execute 语句涉及的表结构在 prepare 语句执行后发生了变化 |
+| 8114 | 未知的执行计划类型 |
+| 8115 | 不支持 prepare 多行语句 |
+| 8116 | 不支持 prepare DDL 语句 |
+| 8117 | 内部 SQL 没有读到指定的系统变量 |
```

Comment:
I doubt this error is caused by system variable, it seems it is because of empty result set, see

```
// getExecRet executes restricted sql and the result is one column.
// It returns a string value.
func (s *session) getExecRet(ctx sessionctx.Context, sql string) (string, error) {
	rows, fields, err := s.ExecRestrictedSQL(sql)
	if err != nil {
		return "", err
	}
	if len(rows) == 0 {
		return "", executor.ErrResultIsEmpty
	}
	d := rows[0].GetDatum(0, &fields[0].Column.FieldType)
	value, err := d.ToString()
	if err != nil {
		return "", err
	}
	return value, nil
}

```

Reply:
I'll remove it, it will be replaced by `ErrUnknownSystemVar` in `GetGlobalSysVar`.